docs/docs/reference/language-versions/binary-compatibility.md

```diff
@@ -0,0 +1,25 @@
+---
+layout: doc-page
+title: "Binary Compatibility"
+---
+
+Thanks to TASTy files, which are produced during compilation of Scala 3 sources, unlike Scala 2, Scala 3 is backward binary compatible between different minor versions (i.e. binary artifacts produced with Scala 3.x can be consumed by Scala 3.y programs as long as x <= y).
+There are however already some ongoing attempts to make Scala 3 forward binary compatible which means that, with some restrictions, it might be possible to compile Scala code with a newer version of the compiler and then use the produced binaries as dependencies for a project using an older compiler.
+
+Scala 3.1.2-RC1 adds an experimental `-Yscala-release <release-version>` compiler flag which makes the compiler produce TASTy files that should be possible to use by all Scala 3 compilers in version `<release-version>` or newer (this flag was inspired by how `-release` works for specifying the target version of JDK). More specifically this flag enforces emitting TASTy files in an older format ensuring that:
```

Comment:
The flag is still experimental so this isn't a big deal yet but when it becomes non-experimental I suggest explicitly mentioning that we do not guarantee that -scala-release will always support all older scala 3 releases (for example, maybe scala 3.5 will only support `-scala-release 3.2` and up), this gives us the option to remove old code to ease our maintenance burden.